packages/frint-cli/commands/new.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-use-before-define */
+
+const mkdirp = require('mkdirp');
+const request = require('request');
+const tar = require('tar');
+
+const createApp = require('frint').createApp;
+
+const DEFAULT_ORGANIZATION = 'Travix-International';
+const DEFAULT_REPOSITORY = 'frint';
+const DEFAULT_BRANCH = 'master';
+const DEFAULT_EXAMPLES_DIR = 'examples';
+const DEFAULT_EXAMPLE = 'counter';
+const DESCRIPTION_TEXT = `
+Usage:
+
+  $ frint new
+  $ frint new <directory>
+  $ frint new <directory> --example <example>
+
+Example:
+
+  $ frint new myapp --example kitchensink
+  $ frint new myapp --example frint-vue/tree/master/examples/basic
+
+You can find a list of all available official examples here:
+https://github.com/Travix-International/frint/tree/master/examples
+`.trim();
+const INVALID_EXAMPLE_ARG_TEXT = `
+Invalid <example> value. Must be in one of the following formats:
+
+  * <name>
+  * <organization>/<repository>/tree/<branch>/**
+`.trim();
+const COMPLETION_TEXT = `
+Done!
+
+Please run these two commands to start your application:
+
+  $ npm install
+  $ npm start
+`.trim();
+
+module.exports = createApp({
+  name: 'new',
+  providers: [
+    {
+      name: 'summary',
+      useValue: 'Scaffolds a new Frint app in specified directory',
+    },
+    {
+      name: 'description',
+      useValue: DESCRIPTION_TEXT,
+    },
+    {
+      name: 'execute',
+      useFactory: function useFactory(deps) {
+        return function execute() {
+          deps.console.log('Initializing...');
+          Promise.resolve(deps)
+            .then(mapDepsToContext)
+            .then(createOutputDirectory)
+            .then(streamExampleToOutputDirectory)
+            .then(() => deps.console.log(COMPLETION_TEXT))
+            .catch(deps.console.error);
+        };
+      },
+      deps: [
+        'console',
+        'params',
+        'pwd',
+      ],
+    }
+  ],
+});
+
+function mapDepsToContext(deps) {
+  return new Promise((resolve, reject) => {
+    // The <example> param has two shapes:
+    // * <name> - example name from the official Frint GitHub repository
+    // * <organization>/<repo>/tree/<branch>/** - full GitHub path to an arbitrary example
+    const example = deps.params.example || DEFAULT_EXAMPLE;
+    if (!example.match(/(^(\w|-)+$)|^\/?(\w|-)+(\/(\w|-)+){3,}\/?$/)) {
+      reject(INVALID_EXAMPLE_ARG_TEXT);
+    }
+
+    const isCustomExample = example.indexOf('/') >= 0;
+    resolve(isCustomExample ? getContextForCustomRepo() : getContextForDefaultRepo());
+
+    function getContextForDefaultRepo() {
```

Comment:
if this function receives `example` as an argument, then it doesn't have to be defined inside a function. they can all be in the root:

```js
function getContextForDefaultRepo(example) {
  return '';
}

function getOutputDirectory(yargs) {
  return '';
}

function getContextForCustomRepo(example) {
  return '';
}

function mapDepsToContext(deps) {
  return '';
}
```

Reply:
Sure, I used nested functions more as a way of grouping rather than for making use of closures. It's a matter of preference, so it should be ideally consistent across the board.

@fahad19 , so you suggest to keep functions flattened within the file root and keep them as pure as possible?

Reply:
yes please. purer the better :)